### PR TITLE
[Samples][13.core-bot] Fix the migration issues

### PR DESCRIPTION
**Pull request by fran893**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/FlightBookingRecognizer.java [hunk 1]
```diff
@@ -41,6 +49,10 @@ public FlightBookingRecognizer(Configuration configuration) {
         }
     }
 
+    /**
+     * Verify is the recognizer is configured.
```
Comment (FedericoBernal) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/FlightBookingRecognizer.java:53]: Typo: it should be "Verify if the recofnizer is configured

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/DialogAndWelcomeBot.java [hunk 2]
```diff
@@ -6,16 +6,26 @@
...
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
+/**
+ * The class which contains the welcome dialog.
+ * @param <T>
```
Comment (Batta32) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/DialogAndWelcomeBot.java:27]:
```suggestion
 * @param <T> is a Dialog
```

**Review by FedericoBernal — CHANGES_REQUESTED** [review]

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java [hunk 3]
```diff
@@ -26,25 +26,34 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import org.apache.commons.lang3.StringUtils;
 
+/**
+ * The class which contains the main dialogs.
```
Comment (Batta32) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java:30]:
```suggestion
 * The class which contains the main dialog for the sample.
```

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/FlightBookingRecognizer.java [hunk 4]
```diff
@@ -41,6 +49,10 @@ public FlightBookingRecognizer(Configuration configuration) {
...
+     * @return True: if is configured, False if it's not.
```
Comment (FedericoBernal) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/FlightBookingRecognizer.java:54]: Typo: it should be "True: if it's configured, False if it's not."

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java [hunk 5]
```diff
@@ -126,27 +140,34 @@ public MainDialog(FlightBookingRecognizer withLuisRecognizer, BookingDialog book
...
 
-        if (StringUtils.isNotBlank(fromEntities.get("from").toString()) && StringUtils.isBlank(fromEntities.get("airport").toString())) {
+        if (StringUtils.isNotBlank(fromEntities.get("from").toString())
+            && StringUtils.isBlank(fromEntities.get("airport").toString())) {
             unsupportedCities.add(fromEntities.get("from").toString());
         }
 
-        if (StringUtils.isNotBlank(toEntities.get("to").toString()) && StringUtils.isBlank(toEntities.get("airport").toString())) {
+        if (StringUtils.isNotBlank(toEntities.get("to").toString())
+            && StringUtils.isBlank(toEntities.get("airport").toString())) {
```
Comment (Batta32) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java:164]: This should be `asText()` instead of `toString()`, right?

**Review by Batta32 — CHANGES_REQUESTED** [review]
nit changes

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java (hunk 5 above)
Comment (fran893) [reply samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java:164]: I'll check it

Comment (fran893) [reply samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/MainDialog.java:164]: You're right. I'll change it

**Review by FedericoBernal — APPROVED** [review]
LGTM

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDetails.java [hunk 6]
```diff
@@ -3,32 +3,60 @@
...
+     */
     public String getTravelDate() {
         return travelDate;
     }
 
-    public void setTravelDate(String travelDate) {
-        this.travelDate = travelDate;
+    /**
+     * Sets the travel date of the booking.
+     * @param withTravelDate The new travel data.
```
Comment (matiasroldan6) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDetails.java:57]: maybe you meant "date"

Comment (fran893) [reply samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDetails.java:57]: Yep you're right, I'll fix it

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java [hunk 7]
```diff
@@ -36,22 +42,22 @@ public BookingDialog() {
             this::confirmStep,
             this::finalStep
         };
-        addDialog(new WaterfallDialog(WaterfallDialog.class.getName(), Arrays.asList(waterfallSteps)));
+        addDialog(new WaterfallDialog("WaterfallDialog", Arrays.asList(waterfallSteps)));
```
Comment (matiasroldan6) [inline samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java:45]: We had this in a constant before, following [TS](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/typescript_nodejs/13.core-bot/src/dialogs/bookingDialog.ts#L20). Not super mandatory though

Comment (fran893) [reply samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java:45]: You're right, but it supposed to be the class name, so i think it's OK the way we have it

**Review by Batta32 — APPROVED** [review]
LGTM

File: samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java (hunk 7 above)
Comment (Batta32) [reply samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java:45]: Agree

Comment (Batta32) [reply samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/BookingDialog.java:45]: We can leave this as is right now

**Pull request merged**